src/game.cpp

```diff
@@ -414,22 +417,57 @@ PointedThing getPointedThing(Client *client, v3f player_position,
 						result.node_abovesurface = np_above;
 						mindistance = distance;
 
-						hilightboxes.clear();
-
-						if (!g_settings->getBool("enable_node_highlighting")) {
-							for (std::vector<aabb3f>::const_iterator
-									i2 = boxes.begin();
-									i2 != boxes.end(); ++i2) {
-								aabb3f box = *i2;
-								box.MinEdge += npf + v3f(-d, -d, -d) - intToFloat(camera_offset, BS);
-								box.MaxEdge += npf + v3f(d, d, d) - intToFloat(camera_offset, BS);
-								hilightboxes.push_back(box);
-							}
+						selectionboxes->clear();
+						for (std::vector<aabb3f>::const_iterator
+								i2 = boxes.begin();
+								i2 != boxes.end(); ++i2) {
+							aabb3f box = *i2;
+							box.MinEdge += v3f(-d, -d, -d);
+							box.MaxEdge += v3f(d, d, d);
+							selectionboxes->push_back(box);
 						}
+						hud->setHighlightedPos(npf, camera_offset);
+						hud->setSelectionMeshScale(v3f(1.0, 1.0, 1.0));
 					}
 				}
 			} // for coords
 
+	if (selectionboxes->size() > 0) {
+		v3f pf = hud->getHighlightedPos();
+		v3s16 p = floatToInt(pf, BS);  
+		// Get selection mesh light level
+		static const v3s16 dirs[7] = {
+			v3s16( 0, 0, 0),
+			v3s16( 0, 1, 0),
+			v3s16( 0,-1, 0),
+			v3s16( 1, 0, 0),
+			v3s16(-1, 0, 0),
+			v3s16( 0, 0, 1),
+			v3s16( 0, 0,-1)
+		};
+		u16 l = 0;
+		u16 l1 = 0;
```

Comment:
Could you use a different variable name than... "l1"?  That's not difficult to read at all or anything </s>